File: solutions/go/isogram/1/isogram.go
// Package isogram solves the Isogram exercism on exercism.
package isogram

import "strings"

// IsIsogram returns true if the given word is an isograme, false otherwise.
func IsIsogram(word string) bool {
	lettersFound := make([]bool, 26)

	for _, c := range strings.ToLower(word) {
		if c < 'a' || c > 'z' {
			continue
		}
		if lettersFound[c-'a'] {
			return false
		}
		lettersFound[c-'a'] = true
	}

	return true
}

File: solutions/fsharp/isogram/1/Isogram.fs
module Isogram

open System

let isIsogram (str: string) =
    let lettersOfString = str.ToLower() |> Seq.filter Char.IsLetter

    lettersOfString
    |> Seq.distinct
    |> Seq.length
    |> (=) (Seq.length lettersOfString)
